pkg/pxc/app/statefulset/node.go

```diff
@@ -144,8 +145,29 @@ func (c *Node) Resources(spec *api.PodResources) (corev1.ResourceRequirements, e
 	return app.CreateResources(spec)
 }
 
-func (c *Node) PVCs(spec *api.VolumeSpec) []corev1.PersistentVolumeClaim {
-	return app.PVCs("datadir", spec)
+func (c *Node) Volumes(cr *api.PerconaXtraDBCluster) *api.Volume {
+	configVolume := app.GetConfigVolumes(cr, c.Lables()["component"])
```

Comment:
Move this into the `if cr.Spec.PXC.Configuration != "" {` body.
1. Logic together 
2. There is no need to create ConfigVolume unless it's specified in config